src/components/Root/Root.tsx

```diff
@@ -6,49 +6,61 @@ import ErrorPage from '#components/ErrorPage/ErrorPage';
-import LoadingOverlay from '#components/LoadingOverlay/LoadingOverlay';
 import DevConfigSelector from '#components/DevConfigSelector/DevConfigSelector';
-import { useConfigSource } from '#src/utils/configOverride';
-import { loadAndValidateConfig } from '#src/utils/configLoad';
-import { initSettings } from '#src/stores/SettingsController';
 import AppRoutes from '#src/containers/AppRoutes/AppRoutes';
 import registerCustomScreens from '#src/screenMapping';
+import LoadingOverlay from '#components/LoadingOverlay/LoadingOverlay';
+// import { initApp } from '#src/modules/init1';
+import { useTrackConfigKeyChange } from '#src/hooks/useTrackConfigKeyChange';
+import { ConfigError, SettingsError } from '#src/utils/error';
+import type { Settings } from '#src/stores/SettingsStore';
+import type { Config } from '#types/Config';
+import { CACHE_TIME, STALE_TIME } from '#src/config';
+import AppController from '#src/stores/AppController2';
+import { getModule } from '#src/modules/container';
+
+type Resources = {
+  config: Config;
+  configSource: string | undefined;
+  settings: Settings;
+};
+
+const applicationController = getModule(AppController);
 
 // This is moved to a separate, parallel component to reduce rerenders
 const RootLoader = ({ setAppIsReady }: { setAppIsReady: React.Dispatch<React.SetStateAction<boolean>> }) => {
   const { t } = useTranslation('error');
-  const settingsQuery = useQuery('settings-init', initSettings, {
-    enabled: true,
-    retry: 1,
+
+  const { data, isLoading, error, isError, isSuccess } = useQuery<Resources, Error>('config-init', async () => applicationController.initApp(), {
     refetchInterval: false,
+    retry: 1,
+    cacheTime: CACHE_TIME,
+    staleTime: STALE_TIME,
   });
 
-  const configSource = useConfigSource(settingsQuery?.data);
+  // Modify query string to add / remove app-config id
+  useTrackConfigKeyChange(data?.settings, data?.configSource);
 
-  const configQuery = useQuery('config-init-' + configSource, async () => await loadAndValidateConfig(configSource), {
-    enabled: settingsQuery.isSuccess,
-    retry: configSource ? 1 : 0,
-    refetchInterval: false,
-  });
+  const { configSource } = data || {};
 
   // After the config loads, we can show the rest of the App
   useEffect(() => {
-    setAppIsReady(!configQuery.isError && !configQuery.isLoading && !!configQuery.data);
-  }, [setAppIsReady, configQuery.isError, configQuery.isLoading, configQuery.data]);
+    setAppIsReady(!isError && !isLoading && !!data);
+  }, [setAppIsReady, isError, isLoading, data]);
 
   const IS_DEMO_OR_PREVIEW = IS_DEMO_MODE || IS_PREVIEW_MODE;
 
   // Show the spinner while loading except in demo mode (the demo config shows its own loading status)
-  if (settingsQuery.isLoading || (!IS_DEMO_OR_PREVIEW && configQuery.isLoading)) {
+  if (!IS_DEMO_OR_PREVIEW && isLoading) {
     return <LoadingOverlay />;
   }
 
-  if (settingsQuery.isError) {
+  if (error?.name === SettingsError.name) {
```

Comment:
Having to have separate error classes and inspect error names indicated to me that there's coupling between settings and config that shouldn't be coupled.

Reply:
I would just consider them to be different types of the init failure. It can be one `InitError` with different payload types. Config service depends on settings to calculate the config id. That is why my first intention was to remove this dependency and move it to the build step.

Reply:
They tie back to different root causes though. Settings errors mean the ini file is missing, broken, or invalid. Config errors may mean the app config service is down or you've asked for a missing or broken config.

Reply:
What do you think about being able to put the title, message, and helplink into the error instead of having error names for different classes that then map back to these different titles, etc.?

Then the only extra logic we need to check for when we have an error is whether to maybe show the demo dialog instead of the error (this is the case when we have a settings object and demo mode is true.)